pkg/api/v1beta3/types.go

```diff
@@ -206,7 +206,10 @@ type HostPathVolumeSource struct {
 	Path string `json:"path" description:"path of the directory on the host"`
 }
 
-type EmptyDirVolumeSource struct{}
+type EmptyDirVolumeSource struct {
+	// Optional: make this empty directory be backed by memory.
+	Tmpfs bool `json:"tmpfs" description:"use tmpfs instead of disk"`
```

Comment:
@bgrant0607 I documented the desired end-state in pkg/api/types/go

Regarding "type" vs "tmpfs" - I made this choice consciously so that we WOULD NOT add arbitrary new types without revisiting the idea of constraint-matching.  I.e. I did this to force this field to be limited-lifetime.

I'll change it to an enum-style "type" if people don't buy that argument.